common/src/main/java/com/wynntils/gui/widgets/SortOrderWidget.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.gui.widgets;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.gui.render.RenderUtils;
+import com.wynntils.gui.render.Texture;
+import com.wynntils.gui.screens.WynntilsQuestBookScreen;
+import com.wynntils.wynn.model.questbook.QuestBookManager;
+import net.minecraft.client.gui.components.AbstractButton;
+import net.minecraft.client.gui.narration.NarrationElementOutput;
+import net.minecraft.network.chat.TextComponent;
+
+public class SortOrderWidget extends AbstractButton {
+    private final WynntilsQuestBookScreen questBookScreen;
+
+    public SortOrderWidget(int x, int y, int width, int height, WynntilsQuestBookScreen questBookScreen) {
+        super(x, y, width, height, new TextComponent("Sort Order Button"));
+        this.questBookScreen = questBookScreen;
+    }
+
+    @Override
+    public void onPress() {
+        QuestBookManager.QuestSortOrder questSortOrder = questBookScreen.getQuestSortOrder();
+
+        QuestBookManager.QuestSortOrder[] questSortOrders = QuestBookManager.QuestSortOrder.values();
+
+        QuestBookManager.QuestSortOrder newSort =
+                questSortOrders[(questSortOrder.ordinal() + 1) % questSortOrders.length];
+
+        questBookScreen.setQuestSortOrder(newSort);
+    }
+
+    @Override
+    public void renderButton(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        Texture sortTexture =
+                switch (questBookScreen.getQuestSortOrder()) {
+                    case LEVEL -> Texture.SORT_LEVEL;
+                    case DISTANCE -> Texture.SORT_DISTANCE;
+                    case ALPHABETIC -> Texture.SORT_ALPHABETICALLY;
+                };
```

Comment:
Maybe store in enum itself?

Reply:
I don't want to associate that enum with this code as the sorting will be used for more stuff.

Reply:
Got it